### PR TITLE
add prometheus ironic exporter

### PR DESCRIPTION
**Pull request by msherman64**

### CODE REVIEW

**Review by Mark-Powers — APPROVED** [review]
👍

**Review by codyhammock — APPROVED** [review]
👍🏻

File: ansible/roles/prometheus/templates/prometheus-ironic-exporter.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[oslo_messaging_notifications]
+# Template this config separately, as the ipmi exporter fails to parse the main ironic.conf
+# in the case that `driver` is defined twice, e.g. for messagingv2 AND prometheus_exporter.
+# Ironic exporter only requires the `location` option.
+driver = prometheus_exporter
+transport_url = fake://
```
Comment (diurnalist) [inline ansible/roles/prometheus/templates/prometheus-ironic-exporter.conf.j2:6]: Does this work out of the box? The general advice is to have the defaults work for K-A things. As little additional operator config as possible.

File: ansible/roles/prometheus/templates/prometheus-ironic-exporter.json.j2 [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+    "command": "gunicorn -b {{ api_interface_address }}:{{ prometheus_ironic_exporter_port }} --env IRONIC_CONFIG=/etc/prometheus/ironic.conf --access-logfile=/var/log/kolla/prometheus/ipe_access.log --error-logfile=/var/log/kolla/prometheus/ipe_errors.log ironic_prometheus_exporter.app.wsgi:application",
```
Comment (diurnalist) [inline ansible/roles/prometheus/templates/prometheus-ironic-exporter.json.j2:2]: Are there any other things using gunicorn? How are other WSGI servers handled in K-A? I thought it mostly used Apache.

**Review by diurnalist — APPROVED** [review]
I think this looks pretty good! I think it would be a good candidate for upstream as well, as it doesn't really depend on any Chameleon-specific things, AFAICT. ⭐

File: ansible/roles/prometheus/templates/prometheus-ironic-exporter.conf.j2 (hunk 1 above)
Comment (msherman64) [reply ansible/roles/prometheus/templates/prometheus-ironic-exporter.conf.j2:6]: It does work out of the box, the exporter side has an extremely minimal configuration, and actually only depends on the line `location = /opt/stack/node_metrics`

**Pull request merged**